src/mesh/region.cc

```diff
@@ -327,8 +321,23 @@ RegionSet RegionDB::union_sets( const string & set_name_1, const string & set_na
 	return set_union;
 }
 
+RegionSet RegionDB::union_sets( RegionSet target_set, const string & source_set_name) const {
```

Comment:
Check set names on call side, pass here reference to the set itself. The the method prepare_set is mere sort and coud be removed. Same for intersection and difference.